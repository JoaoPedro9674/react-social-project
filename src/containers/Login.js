import React, { Component } from 'react';
import { loginWithGithub } from '../backend';

export class Login extends Component {
  constructor(props) {
    super(props);
    this.handleGithubLogin = this.handleGithubLogin.bind(this);
    this.state = {
      hasErrors: false,
      error: null,
    };
  }

  handleGithubLogin(e) {
    e.preventDefault();
    loginWithGithub().catch(err => this.handleAuthError(err));
  }

  handleAuthError(error) {
    this.setState({
      hasErrors: true,
      error,
    });
  }

  render() {
    return (
      <div className="container">
        <div className="row middle-xs text-center">
          <div className="col-xs-12 col-sm-6 col-sm-offset-3">
            <h1>Welcome to Letters Social!
              <br />
            </h1>
            <br />
            <p>
              This is a sample app built with <a href="https://facebook.github.io/react/" rel="noopener noreferrer" target="_blank">React</a> for <em>React in Action</em> by me (Mark Thomas). Feel free to sign in with one of the tools below or use an email and password. The app will never post, email, or do anything without your permission, it is just for learning purposes
            </p>
            <br />
          </div>
        </div>
        <div className="row">
          <div className="col-xs-12 col-sm-6 col-sm-offset-3">
            <form>
              <div className="row center-xs">
                <div className="col-xs-12">
                  <button onClick={this.handleGithubLogin} className="btn btn-default btn-block">
                    <i className="fa fa-github" /> log in with Github
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}
