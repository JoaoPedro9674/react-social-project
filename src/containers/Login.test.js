import React from 'react';
import renderer from 'react-test-renderer';

import { Login } from './Login';

describe('<Login/>', () => {
  it('Login should render correctly', () => {
    const component = renderer.create(<Login />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should ', function () {

  });
});
