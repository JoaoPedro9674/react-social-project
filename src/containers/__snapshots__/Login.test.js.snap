// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> Login should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="row middle-xs text-center"
  >
    <div
      className="col-xs-12 col-sm-6 col-sm-offset-3"
    >
      <h1>
        Welcome to Letters Social!
        <br />
      </h1>
      <br />
      <p>
        This is a sample app built with 
        <a
          href="https://facebook.github.io/react/"
          rel="noopener noreferrer"
          target="_blank"
        >
          React
        </a>
         for 
        <em>
          React in Action
        </em>
         by me (Mark Thomas). Feel free to sign in with one of the tools below or use an email and password. The app will never post, email, or do anything without your permission, it is just for learning purposes
      </p>
      <br />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12 col-sm-6 col-sm-offset-3"
    >
      <form>
        <div
          className="row center-xs"
        >
          <div
            className="col-xs-12"
          >
            <button
              className="btn btn-default btn-block"
              onClick={[Function]}
            >
              <i
                className="fa fa-github"
              />
               log in with Github
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
